src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/ProcessTracker.cs

```diff
@@ -0,0 +1,125 @@
+using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.AspNetCore.Server.IntegrationTesting.IIS
+{
+    // Uses Windows Job Objects to ensure external processes are killed if the current process is terminated non-gracefully.
+    internal static class ProcessTracker
+    {
+        private static readonly IntPtr _jobHandle;
+
+        static ProcessTracker()
```

Comment:
Where did you get this class?

Reply:
Slightly modified version of https://stackoverflow.com/a/37034966/102052